plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextEditorListener.kt

```diff
@@ -0,0 +1,16 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cwc.editor.context.project
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.FileEditorManagerListener
+import com.intellij.openapi.vfs.VirtualFile
+import software.aws.toolkits.jetbrains.services.codewhisperer.settings.CodeWhispererSettings
+
+class ProjectContextEditorListener : FileEditorManagerListener {
+    override fun fileClosed(source: FileEditorManager, file: VirtualFile) {
+        if (CodeWhispererSettings.getInstance().isProjectContextEnabled()) {
+            ProjectContextController.getInstance(source.project).updateIndex(file.path)
```

Comment:
i dont think this should happen if the user is not using chat. closing a file should not cause a server to spin up?

Reply:
If the user has this feature enabled, it's ok to start the server as long as the project is open?

Reply:
maybe it's fine since it's still opt-in, but i would reconsider this behavior